pkg/cmd/pulumi/operations/up.go

```diff
@@ -707,6 +709,9 @@ func NewUpCmd() *cobra.Command {
 		&refresh, "refresh", "r", "",
 		"Refresh the state of the stack's resources before this update")
 	cmd.PersistentFlags().Lookup("refresh").NoOptDefVal = "true"
+	cmd.PersistentFlags().BoolVar(
+		&runProgram, "run-program", env.RunProgram.Value(),
+		"Run the program to determine up-to-date state for providers to refresh resources, this only applies if --refresh is set")
```

Comment:
Should we error out when `--refresh` is not set to not confuse the user?  Not sure it's worth it though.